### ISSUE TITLE
Go: failing to match with `...`

### ISSUE THREAD

**Issue by fcasal** [semgrep/semgrep#9725]
**Describe the bug**
The Go semgrep pattern `var $VAR [...]byte` does not find `var foo [32]byte`

**To Reproduce**
https://semgrep.dev/playground/s/9A45y

**Expected behavior**
The `var $VAR [...]byte` pattern should match with `var foo [32]byte`.

As a workaround, I'm using `$D` instead of `...`.

**What is the priority of the bug to you?**

- [ ] P0: blocking your adoption of Semgrep or workflow
- [x] P1: important to fix or quite annoying
- [ ] P2: regular bug that should get fixed

**Comment by kopecs** [comment]
Is the expected behaviour for you that that pattern would match variable declarations of both slice and array types?

**Comment by fcasal** [comment]
Yes, I expect `...` to match both the empty string in `var foo []byte` and the number in `var foo[32]byte`.
This seems consistent with Semgrep's documentation on the ellipsis operator:

> The ... ellipsis operator abstracts away a sequence of zero or more items such as arguments, statements, parameters, fields, characters.

https://semgrep.dev/docs/writing-rules/pattern-syntax/#ellipsis-operator

**Comment by thescanner42** [comment]
@fcasal I made a scanner that does what you want. e.g. a pattern like `&_[...]` matches either case. [Playground link](<https://thescanner42.github.io/LexerSearchUI/#/RL:)E+B1/d.M+SLVxtAMJV-nGTqyi,Z8HfAB*/V,&kcC!4HV=4e3Tj*L7@ERW-6hq$g>)
